examples/MomentoRedisLambdaExample/MomentoRedisExampleLambdaHandler/src/MomentoRedisExampleLambdaHandler/MomentoCache.cs

```diff
@@ -0,0 +1,188 @@
+        {
+            case CreateCacheResponse.Success:
+                Console.WriteLine($"\t Cache: {_cacheName} created successfully.");
+                break;
+            case CreateCacheResponse.CacheAlreadyExists:
+                Console.WriteLine($"\t Cache: {_cacheName} already exists.");
+                break;
+            case CreateCacheResponse.Error:
+                Console.WriteLine($"\t Error creating cache: {_cacheName}");
+                break;
+        }
+    }
+    
+    public async Task<List<Task<SetResult>>> DoMomentoSets(int numMomentoSetRequests, bool withSleep = false)
+    {
+        Console.WriteLine($"\t WITH SLEEP flag: {withSleep}");
+        try
+        {
+            List<Task<SetResult>> momentoSetTasks = new List<Task<SetResult>>();
+            string sourceData = await Utils.GetSourceData();
+            Console.WriteLine("Executing MomentoSets");
+
+            byte[] compressedSourceData = Compression.Compress(sourceData);
+            
+            for (int i = 1; i <= numMomentoSetRequests; i++)
+            {
+                Console.WriteLine("Setting inside the loop");
+                string key = $"key-{i}";
+                momentoSetTasks.Add(ExecuteSet(key, compressedSourceData, withSleep));
+            }
+
+            return momentoSetTasks;
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine("Exception in set " + e);
+            throw new Exception();
+        }
+    }
+    
+    private async Task<SetResult> ExecuteSet(string key, byte[] value, bool withSleep)
+    {
+        Console.WriteLine($"Beginning set request for {key}");
+        var stopWatch = Stopwatch.StartNew();
+        var setResponseTask = _client.SetAsync(_cacheName, key, value, TimeSpan.FromMinutes(30));
+        Console.WriteLine($"\tIssued set request for {key}");
+        var setResponse = await setResponseTask;
+        Console.WriteLine($"\tDone awaiting set response for {key}");
+        
+        if (withSleep)
+        {
+            await Task.Delay(500);
+            Console.WriteLine($"SET returned from sleeping 500ms for key {key}");
+        }
+        
+        stopWatch.Stop();
+        var duration = stopWatch.ElapsedMilliseconds;
+        Console.WriteLine($"\tReturning set response for {key}: {setResponse}");        
+        return new SetResult(Response: setResponse, ElapsedMillis: duration);
+    }
+    
+#pragma warning disable CS1998
+    public async Task<List<Task<GetResult>>> DoMomentoGets(int numMomentoGetRequests, bool compressValue, bool withSleep = false) {
+#pragma warning restore CS1998
+        List<Task<GetResult>> momentoGetTasks = new List<Task<GetResult>>();
+
+        for (int i = 1; i <= numMomentoGetRequests; i++)
+        {
+            Console.WriteLine("Executing MomentoGets");
+            string key = $"key-{i}";
+            momentoGetTasks.Add(ExecuteGet(key, compressValue, withSleep));
+        }
+
+        return momentoGetTasks;
+    }
```

Comment:
there are no awaits in this function so it doesn't need to be async. maybe that's the warning you disabled above :)

Making this not be async will make the calling code more clear as per my first comment.